server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -685,7 +698,12 @@ public void executeQueryPhase(
                 queryPhase.execute(searchContext);
                 executor.success();
                 readerContext.setRescoreDocIds(searchContext.rescoreDocIds());
-                return new ScrollQuerySearchResult(searchContext.queryResult(), searchContext.shardTarget());
+                ScrollQuerySearchResult scrollQuerySearchResult = new ScrollQuerySearchResult(
```

Comment:
I think we should avoid this and keep the return statement as it was.
something like
```
writeTaskResourceUsage(task);
return new ScrollQuerySearchResult(searchContext.queryResult(), searchContext.shardTarget());
```

Reply:
The idea of this is to do the `writeTaskResourceUsage` at the very end of the search steps to get the total resource usage of the task. So I think it make more sense to also count the resource usage of creating the object?

Reply:
Actually, with the latest change to record failed phases, I'm moving the record logic to `finally` blocks (I should probably have done this before :) ), this should be resolved!